orangecontrib/spectroscopy/io/neaspec.py

```diff
@@ -310,4 +332,298 @@ def handle_data(self, data):
+            )
+        self.domain = np.arange(d_start, d_end, dm)
+        dx = 2 * float(dm) * 1e2  # convert [m] to [cm]
+        self.info["Calculated Datapoint Spacing (Δx)"] = ["[cm]", dx]
+
+    def create_original_df(self, fpath):
+        self.filename = fpath
+        self.info, header_length = self._read_table_metas(self.filename)
+        self.original_df = self._read_table_data(self.filename, header_length)
+        self.original_channel_names = self._extract_channel_names(self.original_df)
+
+    def create_cartesian_df(self):
+        if self.original_df.empty or not self.original_channel_names:
+            raise ValueError("Original data not found")
+
+        self.cartesian_df, self.cartesian_channel_names = self._create_cartesian_df(
+            self.original_df, self.original_channel_names
+        )
+        # clean up the original DataFrame
+        self.original_df = pd.DataFrame()
+
+    def create_resampled_df(self, padding=100):
+        if self.cartesian_df.empty or not self.cartesian_channel_names:
+            raise ValueError("Cartesian data not found")
+        if self.domain.size == 0:
+            raise ValueError("Domain not found")
+        self.resampled_df = self._create_resampled_polar_df_from_cartesian(
+            self.cartesian_df,
+            self.cartesian_channel_names,
+            self.domain,
+            padding=padding,
+        )
+        # clean up the cartesian DataFrame
+        self.cartesian_df = pd.DataFrame()
+
+    def read(self):
+        # builds the spectra table from the output of the read_spectra method
+        out_data_headers, out_data, meta_data = self.read_spectra()
+        features = [
+            ContinuousVariable(name=f"{f}", number_of_decimals=10, compute_value=f)
+            for f in out_data_headers
+        ]
+        domain = Domain(
+            features,
+            class_vars=meta_data.domain.class_vars,
+            metas=meta_data.domain.metas,
+        )
+        return Table.from_numpy(
+            domain, X=out_data, metas=meta_data.metas, attributes=meta_data.attributes
+        )
+
+    def read_spectra(self):
+        self.create_original_df(self.filename)
+        self.create_padded_domain(self.original_df, padding=100)
+        self.create_cartesian_df()
+        self.create_resampled_df(padding=100)
+        df = self.resampled_df
+        # format output to be used in the read method
+
+        self.original_df = []
+        out_data = df.drop(columns=["Row", "Column", "Run", "Channel"]).values
+        out_data = out_data.astype(np.float64)
+        # formatting metas
+        meta_domain = [
+            Orange.data.ContinuousVariable.make("column"),
+            Orange.data.ContinuousVariable.make("row"),
+            Orange.data.ContinuousVariable.make("run"),
+            Orange.data.StringVariable.make("channel"),
+        ]
+        out_meta = df[["Column", "Row", "Run", "Channel"]].values
```

Comment:
Are "Column" and "Row" the XY map positions or something else? Generally we use `map_x` and `map_y` but it will work anyway.

Reply:
we should start using the same x/y notation for position names, so i suggest renaming these to map_x and map_y

I made the same mistake in the "Reshape" widget, but I will change that to conform this nomenclature too

Reply:
Thanks in advance! There were no conventions until now, but I think it is a good time to introduce them... @markotoplak and @stuart-cls, what do you think?

Reply:
I updated that PR to solve just this issue.